src/FixedDataTable.js

```diff
@@ -1056,6 +1056,21 @@ var FixedDataTable = React.createClass({
 
     this._scrollHelper.setViewportHeight(bodyHeight);
 
+    // This calculation is synonymous to Element.scrollTop
+    var scrollTop = firstRowIndex && Math.abs(firstRowOffset - this._scrollHelper.getRowPosition(firstRowIndex));
+    if (scrollTop >= 0) {
+      var maxScrollTop = maxScrollY ? scrollContentHeight - bodyHeight : 0;
+      // Handle the case where the scrollTop is beyond the maxScrollTop, such as when the user is completely scrolled
+      // down and resizes the viewport to be smaller vertically
+      if (scrollTop > maxScrollTop || maxScrollTop === 0) {
```

Comment:
Got it! Originally I couldn't come up with a way to do the maxScrollTop === 0 once. Before, if the 0 logic was removed, scrolling slowly back up can cause the scrollTop to never update all the way back to 0. At the time I couldn't figure out how to do the check so that we only set back to 0 once.

But I found out how to do it. I have the previous scrollY, so:
`if (scrollTop > maxScrollY || (oldState && oldState.scrollY !== 0 && maxScrollY === 0)) {`

See in my newest commit.